src/contrasts.jl

```diff
@@ -160,11 +160,11 @@ function ContrastsMatrix(contrasts::AbstractContrasts, levels::AbstractVector)
     end
 
     # find index of base level. use contrasts.base, then default (1).
-    baseind = isnull(contrasts.base) ?
+    baseind = contrasts.base === nothing ?
```

Comment:
How about `contrasts.base isa Nothing`?

Reply:
In general it gives the same code. I think the exception is when the return type depends on this condition, in which case `isa` is currently the only way to make it inferrable.